### PR TITLE
Implement basic front-end layout

### PR DESCRIPTION
**Pull request by nvandoorn**
Front-end implementation is still **very** far from complete, but this PR addresses some pretty major improvements.

# Changes in this PR

1. Implement a functional webpack css pipeline (css-js-loader and css-modules for dev and prod)
2. Implement Basic front-end styling designed to mimic the Sketch mockups
3. Refactor/reorganize of `/src/client`
4. Remove un-used files (and comments in `/public/index.html`)
5. Fix server-side enum truthy bug (`const s = constants.levels.error ? statement1 : statement2` was causing issues due to `constants.levels.error = 0`)
6. Update readme for the changes mentioned above

# Known Deficiencies
1. Directory listing sidebar from Sketch mockups is not implemented yet
2. No front-end tests
3. CSS globals export/import syntax is inconsistent 
4. `case` statement used to map level to css module is not very terse
5. Front end request handling is janky (considering solving this with Redux -- I'm really not sure if the overhead is worth it here)

### CODE REVIEW

File: src/client/helpers/http.js [hunk 1]
```diff
@@ -8,13 +8,14 @@ function httpReq(type, url, body){
     xhr.setRequestHeader('Content-Type', 'application/json');
     xhr.send(JSON.stringify(body));
     xhr.onreadystatechange = function(){
-      if(this.readyState === 4 && this.status === 200) {
```
Comment (jordanmckinney) [inline src/client/helpers/http.js]: What does 4 signify? Would it make sense to change this to a constant?

File: src/constants.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  debug: 4
+};
+
+exports.defaultLevel = exports.levels.debug;
+
+exports.defaultLevelStr = 'debug';
+
+exports.defaultPageSize = 20;
+
+exports.defaultDatetimePattern = /\[(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z)\]/;
```
Comment (jordanmckinney) [inline src/constants.js]: Not sure if there is a standard ISO datetime pattern that could be used, or whether that would make sense here.

File: src/scss/stylesheets/style.scss [hunk 3]
```diff
@@ -1,91 +0,0 @@
...
-    text-decoration: none;
-    font-size: 1.1em;
-  }
-
-  a:hover{
-    color: rgb(252, 161, 88);
-  }
-}
-
-
```
Comment (jordanmckinney) [inline src/scss/stylesheets/style.scss]: Nitpick: was 4 blank lines intentional?

File: src/scss/stylesheets/style.scss [hunk 4]
```diff
@@ -1,91 +0,0 @@
...
-#loglines{
-  animation-duration: 3s;
-  animation-name: slidein;
-  border-radius: 6px;
-  background: $rounded-bg;
-  padding: 20px;
-  overflow-y: scroll;
-  height: calc(100vh - 200px);
-  background: rgb(38, 42, 50);
-
```
Comment (jordanmckinney) [inline src/scss/stylesheets/style.scss]: Intentional 2 blank lines?

File: src/constants.js (hunk 2 above)
Comment (nvandoorn) [inline src/constants.js]: I believe that pattern should match `2017-04-02T20:01:14Z` (ISO 8601)

File: src/scss/stylesheets/style.scss (hunk 3 above)
Comment (nvandoorn) [inline src/scss/stylesheets/style.scss]: Whitespace definitely was not intentional, but everything in `/src/scss` is removed in this PR.

File: src/scss/stylesheets/style.scss (hunk 4 above)
Comment (nvandoorn) [inline src/scss/stylesheets/style.scss]: Please see above.

File: src/client/helpers/http.js (hunk 1 above)
Comment (nvandoorn) [inline src/client/helpers/http.js]: You're right this should be an enum. I'll push another commit to fix this.

Comment (nvandoorn) [inline src/client/helpers/http.js]: Fixed in eca9100

**Comment by nvandoorn** [comment]
Thanks @jordanmckinney for pointing out the magic numbers in the XHR code.

**Comment by nvandoorn** [comment]
For some reason the ES6 `import` and `export` keyword don't work in a96fe85. Spent a good couple hours trying to figure out why but I still have no idea.

**Pull request merged**